python/cugraph/cugraph/dask/structure/mg_property_graph.py

```diff
@@ -331,6 +331,34 @@ def add_vertex_data(self,
                        for n in self.__vertex_prop_dataframe.columns])
         self.__vertex_prop_eval_dict.update(latest)
 
+    def get_vertex_data(self, vertex_ids=None, types=None, columns=None):
+        """
+        Return a dataframe containing vertex properties for only the specified
+        vertex_ids, columns, and/or types, or all vertex IDs if not specified.
+        """
+        if self.__vertex_prop_dataframe is not None:
+            if vertex_ids is not None:
+                df_mask = self.__vertex_prop_dataframe[self.vertex_col_name]\
+                              .isin(vertex_ids)
+                df = self.__vertex_prop_dataframe.loc[df_mask]
+            else:
+                df = self.__vertex_prop_dataframe
+
+            if types is not None:
+                # FIXME: coerce types to a list-like if not?
+                df_mask = df[self.type_col_name].isin(types)
+                df = df.loc[df_mask]
+
+            if columns is None:
+                # Note: this includes the "internal" pG.vertex_col_name and
+                # pG.type_col_name columns, since they are assumed to be needed
+                # by the caller.
+                columns = self.__vertex_prop_dataframe.columns
+                df = df[columns]
```

Comment:
Is this the correct indentation?  Did you mean to dedent it so it always runs?  For example, what happens if the user-provided `columns` is not None?

Reply:
Thank you for noticing this. It's definitely a bug, and it also exposed a problem in my tests since they didn't catch it.
I updated my tests, verified that they caught this problem, updated the code here and in `get_edge_data` since it also had the same problem, then verified that the new tests passed.